### DISCUSSION TITLE
JS Library for Syntax Highlighting?

### DISCUSSION THREAD

**Discussion by Gregoor** [tree-sitter/tree-sitter#851] [Q&A] [answered]
Heya,

still loving tree-sitter, and while I'm not developing my current prototype in Rust anymore, I'm strongly considering using it for syntax highlighting in my webby TS prototype (all those regex-highlighters are broken too easily). Now I'm wondering whether there is already some prior work in making a simple (dumbed-down) library for syntax highlighting purposes only? Otherwise I'd start hacking on that.

Cheers and thanks for the hard work on this great lib!

**Comment by razzeee** [comment, marked as answer]
There is syntax highlighting or is that not enough for you?
https://tree-sitter.github.io/tree-sitter/syntax-highlighting

I think the wasm bindings for this are not very complete

**Reply by Gregoor** [reply]
Oh for some reason I assumed [web-tree-sitter](https://www.npmjs.com/package/web-tree-sitter) wouldn't support that, because it looked like it was in a [separate module](https://github.com/tree-sitter/tree-sitter/tree/master/highlight)? And the whole query bit made me wonder if there's something more immediately consumable.
But I guess if that is support in the npm package, it's just a matter of creating a less-configurable simpler API for that. Cool! Thank you

**Comment by stereobooster** [comment]
There is also this project https://github.com/s0/tree-sitter-hast, which may be interesting to you
